src/java/org/apache/cassandra/service/StartupChecks.java

```diff
@@ -357,14 +357,26 @@ public void execute(StartupChecksOptions options) throws StartupException
         }
     };
 
-    public static final StartupCheck initSigarLibrary = new StartupCheck()
+    public static final StartupCheck checkProcessEnvironment = new StartupCheck()
     {
         @Override
         public void execute(StartupChecksOptions options)
         {
             if (options.isDisabled(getStartupCheckType()))
                 return;
-            new SystemInfo().warnIfRunningInDegradedMode();
+
+            Optional<String> degradations = new SystemInfo().isDegraded();
+
+            if (degradations.isPresent())
+            {
+                String message = "Cassandra server running in degraded mode. " + degradations.get();
+                if (CassandraRelevantProperties.TEST_IGNORE_PROCESS_ENVIRONMENT_CHECK.getBoolean() || CassandraRelevantProperties.TEST_IGNORE_PROCESS_ENVIRONMENT_CHECK.getBoolean())
```

Comment:
@Claudenw there are two same checks in this if

Reply:
@Claudenw ah yes, I see that. Good point. The original code logs the message even there are no degradations. I do not see any value in it. I think this should be just simplified like 

`if (degradations.isPresent) logger.warn(message) else logger.info("optimal performance")`

Reply:
python dtests are not checking it either. +1 on the flag removal